src/portal/mod.rs

```diff
@@ -195,6 +206,44 @@ impl Keyring {
         self.write().await
     }
 
+    /// Replaces item at the given index.
+    ///
+    /// The `index` refers to the index of the [`Vec`] returned by
+    /// [`items()`](Self::items). If the index does not exist, the functions
+    /// returns an error.
+    pub async fn replace_item_index(&self, index: usize, item: &Item) -> Result<(), Error> {
```

Comment:
Not a fan of the name of those two methods but I don't have a better suggestion neither

Reply:
Me neither